TestFoundation/TestCodable.swift

```diff
@@ -33,35 +33,42 @@ private func makePersonNameComponents(namePrefix: String? = nil,
+                                          encode: (T) throws -> Data,
+                                          decode: (Data) throws -> T) throws where T : Equatable  {
     do {
         let data = try encode(value)
         let decoded: T = try decode(data)
         if value != decoded {
-            throw NSError(domain: "Decode mismatch", code: 0, userInfo: ["msg": "Decoded \(T.self) <\(decoded)> not equal to original <\(value)>"])
+            let errorMessage = "Decoded \(T.self) <\(decoded)> not equal to original <\(value)>"
+            throw NSError(domain: "Decode mismatch", code: 0, userInfo: [NSLocalizedDescriptionKey: errorMessage])
         }
     }
 }
 
-func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T) throws where T : Equatable {
-    let inf = "INF", negInf = "-INF", nan = "NaN"
-    let encode = { (_ value: T) throws -> Data in
-        let encoder = JSONEncoder()
-        encoder.nonConformingFloatEncodingStrategy = .convertToString(positiveInfinity: inf,
-                                                                      negativeInfinity: negInf,
-                                                                      nan: nan)
-        return try encoder.encode(value)
-    }
+func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T, lineNumber: UInt) where T : Equatable {
+    do {
+        let inf = "INF", negInf = "-INF", nan = "NaN"
+        let encode = { (_ value: T) throws -> Data in
+            let encoder = JSONEncoder()
+            encoder.nonConformingFloatEncodingStrategy = .convertToString(positiveInfinity: inf,
+                                                                          negativeInfinity: negInf,
+                                                                          nan: nan)
+            return try encoder.encode(value)
+        }
 
-    let decode = { (_ data: Data) throws -> T in
-        let decoder = JSONDecoder()
-        decoder.nonConformingFloatDecodingStrategy = .convertFromString(positiveInfinity: inf,
-                                                                        negativeInfinity: negInf,
-                                                                        nan: nan)
-        return try decoder.decode(T.self, from: data)
-    }
+        let decode = { (_ data: Data) throws -> T in
+            let decoder = JSONDecoder()
+            decoder.nonConformingFloatDecodingStrategy = .convertFromString(positiveInfinity: inf,
+                                                                            negativeInfinity: negInf,
+                                                                            nan: nan)
+            return try decoder.decode(T.self, from: data)
+        }
 
-    try expectRoundTripEquality(of: value, encode: encode, decode: decode)
+        try expectRoundTripEquality(of: value, encode: encode, decode: decode)
+    } catch {
+        XCTFail("\(error)", line: lineNumber)
+    }
```

Comment:
Why is using XCTFail here directly instead of just propagating the error out a better approach?

Reply:
To avoid repetitive `do { … } catch { XCTFail("\(error)", line: lineNumber) }` in each unit test.
With this `expectRoundTripEqualityThroughJSON` behaves like an `XCTAssert`.

Unless I misunderstood you question.